stingray/crossspectrum.py

```diff
@@ -445,16 +452,17 @@ def _make_segment_spectrum(self, lc1, lc2, segment_size):
         if lc1.tseg != lc2.tseg:
             raise ValueError("Lightcurves do not have same tseg.")
 
-        # number of bins per segment
-        nbins = int(segment_size/lc1.dt)
-        start_ind = 0
-        end_ind = nbins
+        if gti is None:
+            gti = cross_two_gtis(lc1.gti, lc2.gti)
```

Comment:
should we not assume that any GTIs are in `lc1` and `lc2`, rather than allowing the user to put extra GTIs in? If we do, it might make things more muddled, because it's not clear whether that will mean any GTIs in `lc1` and `lc2` will be ignored, or whether the new GTIs will be added to those that are already there. 

I'd probably keep GTIs in a single place; as in, if someone wants to use GTIs, they ought to be defined in the `Lightcurve` object, but if you have a compelling usecase, I'm happy to change my mind.